api/v1beta2/watcher_types.go

```diff
@@ -23,12 +23,15 @@ import (
 
 // WatcherSpec defines the desired state of Watcher.
 type WatcherSpec struct {
-	// ServiceInfo describes the service information of the operator
+	// ServiceInfo describes the service information of the listener
 	ServiceInfo Service `json:"serviceInfo"`
 
 	// LabelsToWatch describes the labels that should be watched
 	LabelsToWatch map[string]string `json:"labelsToWatch"`
 
+	// ResourceToWatch is the GroupVersionResource of the resource that should be watched.
+	ResourceToWatch WatchableGVR `json:"resourceToWatch"`
```

Comment:
I am not sure here if it is okay to introduce a breaking change to an existing API version

Reply:
As the watcher API is not released yet. It is okay to introduce a breaking change to its API.